src/core/thread/dua_manager.cpp

```diff
@@ -623,8 +621,11 @@ Error DuaManager::ProcessDuaResponse(Coap::Message &aMessage)
             mDuaState             = kRegistered;
             break;
         case ThreadStatusTlv::kDuaReRegister:
-            mDuaState                  = kToRegister;
-            mDelay.mFields.mCheckDelay = Mle::kImmediateReRegisterDelay;
+            if (Get<ThreadNetif>().HasUnicastAddress(GetDomainUnicastAddress()))
+            {
+                RemoveDomainUnicastAddress();
+                AddDomainUnicastAddress();
```

Comment:
Explain: Removing and adding the DUA again triggers both DUA.req and `Child Update Request` (for MTD).